nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+
+Additionally, since there are a significant amount of overlap in the protocols between NEO and Ontology for sending assets, and interacting with contract on NeoVM, this proposal is a joint effort with [https://github.com/ontio/OEPs/pull/8 OEP6]. Since there are differences in the two platforms, the result of these proposals will not be identical, nor should they be, but getting as much overlap as possible will help to simplify cross-chain interactions for both platforms.
+
+
+==Specification==
+
+'''Provider Request Handling'''
+
+Implementation details per dApp domain handling will be left to the discretion of the wallet provider. Providers can choose to allow users to trust certain dApps to automatically execute transaction on their behalf, or a certain subset of transaction types. Below are basic guidelines for general handling of requests from dApps via the protocol interface.
+
+===Read Methods===
+
+====getProvider====
+Returns information about the  dAPI provider, including who this provider is, the version of their dAPI, and the NEP that the interface is compatible with.
+
+
+'''Method Interface'''
+
+<pre>
+function getProvider(): Promise<Provider>
+</pre>
+
+
+'''Input arguments'''
+
+None
+
+
+'''Success return value'''
+
+object - Provider information defined by the interface below:
+<pre>
+interface Provider {
+  name: string;
+  website: string;
+  version: string;
+  compatibility: string[];
+  extra: object; //this object can contain any attributes specific to the dapi provider, such as an app theme.
+}
+</pre>
+
+The compatibility field will return a list of strings that corresponds to NEPs that the dAPI conforms to. For example:
+<pre>
+compatibility: [
+    'NEP-14',
+    'NEP-23',
+    'NEP-29'
+]
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getProvider()
+.then((provider: Provider) => {
+  const {
+    name,
+    website,
+    version,
+    compatibility,
+    extra,
+  } = provider;
+
+  console.log('Provider name: ' + name);
+  console.log('Provider website: ' + website);
+  console.log('Provider dAPI version: ' + version);
+  console.log('Provider dAPI compatibility: ' + JSON.stringify(compatibility));
+  console.log('Extra provider specific atributes: ' + JSON.stringify(compatibility));
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp.');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+* The name of the provider wallet
+* The provider website
+* The provider specific version of the wallet
+* A list of NEP for which the specific provider protocol is compatible with
+
+Example
+<pre>
```

Comment:
Can `<pre>` be replaced by github markdown triple backticks? So json is better formatted here.
```js
int x = 10; // hello
```

Reply:
I would also very much prefer using markdown over mediawiki for these proposals. However since it is mediawiki, the backticks don't come out looking very good:

![image](https://user-images.githubusercontent.com/6342442/62435985-13a4bd00-b779-11e9-8d94-aa873f497570.png)

Reply:
Isnt there another way to do language formatting here? If not,shouldnt we simply migrate to markdown instead of mediawiki @erikzhang? Code formatting may not be fundamental, but its very important.